dart/lib/src/http_client/failed_request_client.dart

```diff
@@ -0,0 +1,174 @@
+import 'package:http/http.dart';
+import '../protocol.dart';
+import '../hub.dart';
+import '../hub_adapter.dart';
+
+/// A [http](https://pub.dev/packages/http)-package compatible HTTP client
+/// which records events for failed requests.
+///
+/// Configured with default values, this captures requests which throw an
+/// exception.
+/// This can be for example for the following reasons:
+/// - In an browser environment this can be requests which fail because of CORS.
+/// - In an mobile or desktop application this can be requests which failed
+///   because the connection was interrupted.
+///
+/// Additionally you can configure specific HTTP response codes to be considered
+/// as a failed request. In the following example, the status codes 404 and 500
+/// are considered a failed request.
+///
+/// ```dart
+/// import 'package:sentry/sentry.dart';
+///
+/// var client = FailedRequestClient(
+///   failedRequestStatusCodes: [404, 500]
```

Comment:
wondering if it makes sense to be a range, instead of a List of codes, like 400..599

Reply:
Yeah, that's a good idea